src/components/AppWindow/MenuComponents/MenuModal.tsx

```diff
@@ -1,30 +1,66 @@
-import React from 'react'
+import React, { useEffect, useCallback } from 'react'
 import styled from 'styled-components'
 
+interface Props {
+    open: boolean
+    onClose: () => void
+}
+
+export const MenuModal = ({open, onClose}: Props) => {
+    const modalRef = React.createRef<HTMLInputElement>();
+
+
+    const onClickOutside = useCallback(
+        (e: Event) => {
+            if (modalRef.current && modalRef.current.contains(e.target as Node)) {
+                return;
+            }
+            if (open) return onClose()
+        }, [modalRef, onClose, open]
+    )
+
+    useEffect(() => {
+      if (open)  {
+        document.addEventListener("click", onClickOutside);
+      } 
+
+        return () => {
+            document.removeEventListener("click", onClickOutside);
+          };
+    }, [onClickOutside, open]) 
```

Comment:
you can create useOnClickOutside hook 
here example https://usehooks.com/useOnClickOutside/

Reply:
Totally agree!